docs/docs_cpp_code_guidelines.md

```diff
@@ -91,6 +93,23 @@ Prefer `int32_t` over `int`, `int16_t` over `short`, etc.
 #endif // _DEBUG
 ```
 
+### Using
+
+First of all, prefer not to use `using` at all. Especially with STL - containers and so on. Use `class` or to encapsulate something.
+
+Always use `using` instead of `typedef`:
```

Comment:
For class member types, that's not possible (unless it's `using` for some members/methods of parent class), e.g. following doesn't work:
```c++
class Foobar
{
public:
  using SomeIdType = int64_t;
  using CallbackFunction = std::function<bool(SomeIdType)>;
  using RawCallback = bool(*)(int);
};
```
So IMO this should be explicitly stated that in such cases `using` outside of class should be used. Or, probably, allow `typedef` in these cases (maybe except raw function pointers, i.e. `SomeIdType` and `CallbackFunction` from the above example)